src/routes/safe/components/Apps/components/ConfirmTxModal/ReviewConfirm.tsx

```diff
@@ -98,7 +98,8 @@ export const ReviewConfirm = ({
     [txs, isMultiSend],
   )
   const txValue: string | undefined = useMemo(
-    () => (isMultiSend ? '0' : txs[0]?.value && parseTxValue(txs[0]?.value)),
+    // Value is converted to string because numbers were allowed in the safe-apps-sdk v1, so 0 and anything would evaluate as false
+    () => (isMultiSend ? '0' : txs[0]?.value.toString() && parseTxValue(txs[0]?.value)),
```

Comment:
What happens if `value` is `undefined`?

Reply:
If you're expecting `undefined`, I would suggest optionally chain everything: `txs?.[0]?.value?.toString()`.

Reply:
https://github.com/gnosis/safe-apps-sdk/blob/4ce9a16b2d8f4133f6a7e7fbc25f422210cba786/src/types.ts#L35 the type signature doesn't support `undefined`, actually, it doesn't even support numbers 😅 

Currently, the value would be set to '0'. I think app developers should test their apps, if we'll account for each possible type things will get crazy.

Reply:
@mikheevm but how it is currently this will crash the whole web app, right?

Reply:
I mean I am fine that we error, but then we should properly check what is send to the interface. I would also expect to see tests for something like this.

Reply:
> if we'll account for each possible type things will get crazy

I tend to agree but we don't have to check for wrong types, we need to check for the expected types.

In this case `if (typeof value !== 'string') { throw new CodedException('Wrong type used for value') }`.

Maybe we could include some runtime type-checker for the apps API, similar to PropTypes in React? And do it in a declarative fashion?

Reply:
we have a validation method that checks exactly this: https://github.com/gnosis/safe-react/blob/dev/src/routes/safe/components/Apps/components/ConfirmTxModal/index.tsx#L26

Reply:
Found this: https://github.com/codemix/babel-plugin-typecheck
And this: https://github.com/fabiandev/ts-runtime

Reply:
Then undefined should never get that far 👍